keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resolvers/ImageResolver.kt

```diff
@@ -44,46 +47,46 @@ class ImageResolver(
     val imageProvider = resource.spec.imageProvider ?: return resource
     val image = runBlocking {
       when (imageProvider) {
-        is ArtifactImageProvider -> resolveFromArtifact(resource, imageProvider)
+        is ReferenceArtifactImageProvider -> resolveFromReference(resource, imageProvider)
+        // todo eb: artifact provider is here for backwards compatibility. Remove?
+        is ArtifactImageProvider -> resolveFromArtifact(resource, imageProvider.deliveryArtifact as DebianArtifact)
         is JenkinsImageProvider -> resolveFromJenkinsJob(resource, imageProvider)
       }
     }
     return resource.withVirtualMachineImages(image)
   }
 
+  private suspend fun resolveFromReference(
```

Comment:
this is the bit that looks up the correct artifact given a reference in the ec2 cluster spec